File: backend/app/schemas/investigation.py
from pydantic import BaseModel, Field
from typing import List, Optional, Any, Dict

class InvestigationCreate(BaseModel):
    question: Optional[str] = None
    demo_preset_id: Optional[str] = None
    category: Optional[str] = None

class InvestigationQuerySchema(BaseModel):
    query: str
    engine: str
    purpose: str
    priority: int = 1

class InvestigationResponse(BaseModel):
    id: str
    title: str
    original_question: str
    category: str
    status: str
    queries: Optional[List[Dict[str, Any]]] = None
    evidence_count: int = 0
    contradiction_count: int = 0
    report: Optional[Dict[str, Any]] = None
    created_at: str
    completed_at: Optional[str] = None

File: src/lib/db/models/Investigation.ts
import mongoose, { Schema, Document } from 'mongoose';
import { InvestigationCategory, InvestigationStatus } from '@/types';

export interface IInvestigation extends Document {
  title: string;
  original_question: string;
  category: InvestigationCategory;
  status: InvestigationStatus;
  queries: Array<{
    query: string;
    engine: string;
    purpose: string;
    priority?: number;
  }>;
  evidence_count: number;
  contradiction_count: number;
  report?: any;
  events?: any[];
  created_at: Date;
  completed_at?: Date;
}

const InvestigationSchema = new Schema<IInvestigation>({
  title: { type: String, required: true },
  original_question: { type: String, required: true },
  category: { type: String, required: true },
  status: { type: String, required: true, default: 'queued' },
  queries: [
    {
      query: { type: String, required: true },
      engine: { type: String, required: true },
      purpose: { type: String, required: true },
      priority: { type: Number, default: 1 }
    }
  ],
  evidence_count: { type: Number, default: 0 },
  contradiction_count: { type: Number, default: 0 },
  report: { type: Schema.Types.Mixed, default: null },
  events: [{ type: Schema.Types.Mixed, default: [] }],
  created_at: { type: Date, default: Date.now },
  completed_at: { type: Date, default: null }
});

export const InvestigationModel =
  mongoose.models.Investigation || mongoose.model<IInvestigation>('Investigation', InvestigationSchema);
